.github/workflows/build-main.yml

```diff
@@ -13,6 +13,25 @@ env:
   MAVEN_VERSION: '3.9.2'
 
 jobs:
+  setup:
+    name: Setup
+    runs-on: ubuntu-22.04
+    outputs:
+      timestamp: ${{ steps.get-date.outputs.date }}
+      setupSuccessful: "true"
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Get current date
+        id: get-date
+        run: echo "::set-output name=date::$(date +'%Y-%m-%d %H:%M:%S')"
+
+      - name: Get Latest Merge Commit SHA
+        id: get-sha
+        run: |
+          latest_merge_sha=$(git rev-parse origin/master)
+          echo "::set-output name=latestMergeSha::$latest_merge_sha"
```

Comment:
FYI set-output commands are being deprecated:
https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/